### PR TITLE
Draft installation and setup

### PR DESCRIPTION
**Pull request by lsgunnlsgunn**

### CODE REVIEW

File: docs/smart-contracts/install.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  Pact Core enables sustainable growth of Pact features within the Kadena ecosystem by offering more modular and maintainable internal structures to better support community development and enhancements to the language and components that rely on it.
+  For more information about Pact Core, see the following resources:
+
+  - [Pact Core repository](https://github.com/kadena-io/pact-5)
+  - [Pact 5 Beta: Improving Developer, Miner, and User Experience](https://www.kadena.io/perspectives/pact-5-beta)
+
+## Installation options
+
+You can download and install the Pact programming language and interactive interpreter locally on your local computer from prebuilt platform-specific binaries or build Pact directly from its source code. 
+You can also run Pact in a browser without installing it as a component in the development network Docker image.
```
Comment (rsoeldner) [inline docs/smart-contracts/install.md:46]: This is misleading. Pact-5 has no _web_ version to interact with. A very old pact version is part of Chainweaver but it's also outdated. I would remove this.

File: docs/smart-contracts/install.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+The installation instructions in this section describe how to install Pact locally using prebuilt binaries for each operating system platform.
+
+To build Pact binaries directly from source, download the source code from [Pact Releases](https://github.com/kadena-io/pact/releases), then use Homebrew, Cabal from the Haskell toolchain, or the Nix package manager to build Pact.
```
Comment (rsoeldner) [inline docs/smart-contracts/install.md:50]: This is a tricky one. Pact-5 lives in a different repository: https://github.com/kadena-io/pact-5

Also, building from sources, users need a Haskell toolchain that can be brought into scope by using Nix. The instructions on how to do so are currently here: https://github.com/kadena-io/pact-5?tab=readme-ov-file#installing-pact-core

File: docs/smart-contracts/install.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Installation options
+
+You can download and install the Pact programming language and interactive interpreter locally on your local computer from prebuilt platform-specific binaries or build Pact directly from its source code. 
+You can also run Pact in a browser without installing it as a component in the development network Docker image.
+
+The installation instructions in this section describe how to install Pact locally using prebuilt binaries for each operating system platform.
+
+To build Pact binaries directly from source, download the source code from [Pact Releases](https://github.com/kadena-io/pact/releases), then use Homebrew, Cabal from the Haskell toolchain, or the Nix package manager to build Pact. 
+For more information about the dependencies and tools for building from the source code, see [Building from source](https://github.com/kadena-io/pact?tab=readme-ov-file#building-from-source).
```
Comment (rsoeldner) [inline docs/smart-contracts/install.md:51]: Pact < 5 has different build instructions. I'm not sure if we still want to highlight these as we will not maintain the codebases in the long run

File: docs/smart-contracts/install/local-dev-node.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
+---
+title: Local development network
+id: local-dev-node
+sidebar_position: 5
+description: "Start a local development network with a standalone blockchain node."
+---
+
+# Set up a development network
+
+Computers that run the Chainweb consensus protocol—the `chainweb-node` software—are a crucial component of the Kadena blockchain.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:10]: Chainweb is not just consensus, it also provides the P2P network, mempool, etc. Maybe @chessai or Lars can comment on this here?

File: docs/smart-contracts/install/local-dev-node.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+Nodes provide the compute, storage, and network resources that enable the Kadena blockchain to be secure and decentralized.
+Nodes also provide the engine for transaction processing, data verification, and executing smart contract functions.
+Because of the role they play in receiving and processing transactions, it's important to include at least one standalone node as part of your development environment.
+
+In most cases, you can run a single node to establish a local development network that you can then use to test smart contract operations in an isolated and controlled environment before deploying to a public test or production network.
+
+There are two primary ways you can set up a local development network with the `chainweb-node` program:
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:17]: Maybe something like:
```suggestion
There are two primary ways you can set up a local development network that includes the `chainweb-node` program:
```
?

File: docs/smart-contracts/install/local-dev-node.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+- You can download a complete local development network that includes the `chainweb-node` program, a local block explorer, the Pact interpreter that runs in a browser, and other tools. 
+  With this option, you can use a default set of tools and configuration settings from a Docker image to run the development network inside of a Docker container.
+  For instructions, see [Start the development network](#start-the-development-network).
+- You can download and run the `chainweb-node` program as a standalone release binary from its GitHub repository.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:22]: I think officially we provide Docker images, that include the chainweb-node binary. @chessai is this correct?

File: docs/smart-contracts/install/local-dev-node.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+   All actions you perform in Chainweaver are now executed on the local development network.
+
+## Install a standalone node
+
+If you want full control over node operations, you might want to install and manage `chainweb-node` as a standalone local node for development or as a node participating in the Kadena test or production public network.
+
+There are several options for setting up a Chainweb node in a physical or virtual environment.
+For example, you can install and run node binaries:
+
+- Directly on a physical Linux or macOS computer.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:128]:
```suggestion
- Directly on a physical Linux.
```

To my knowledge, currently ARM CPUs are not supported (which also includes pact)

File: docs/smart-contracts/install/local-dev-node.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+- On an Linux or macOS instance with infrastructure from a cloud services provider.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:129]:
```suggestion
- On an Linux with infrastructure from a cloud services provider.
```

File: docs/smart-contracts/install/local-dev-node.md [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+- From a Linux or macOS image in a Docker container.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:130]:
```suggestion
- From a Linux image in a Docker container.
```

File: docs/smart-contracts/install/local-dev-node.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+- Operating system: Linux AMD64 architecture.
+
+If you also plan to use the node for mining, to submit transactions, or to query blockchain data, you should verify that your computer meets the following minimum recommended hardware requirements: 
+
+- Four (4) CPU cores.
+- Eight (8) GB of RAM.
+
+#### Operating system architecture
+
+Most `chainweb-node` software runs on Linux or macOS with AMD64 architectures.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:151]:
```suggestion
Most `chainweb-node` software runs on Linux with AMD64 architectures.
```

File: docs/smart-contracts/install/local-dev-node.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+If you want to run a Chainweb node on Linux or macOS with ARM64 architecture, you'll need to build the binaries yourself from the source code.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:152]: The macOS changes have to be done in the entire document

File: docs/smart-contracts/install/local-dev-node.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+You might experience some unexpected behavior when running a node on ARM64 architecture.
+Note that you should not configure **mining** for nodes on ARM64 architecture.
+RUnning a mining node on ARM64 architecture could cause your node to fork from the network.
+
+If you want to run a Chainweb node on Microsoft Windows, you should note that this isn't a supported configuration and requires you to install the latest Windows Services for Linux (WSL) software.
+If you choose to proceed after installing WSL, you can install the `chainweb-node` software in the WSL virtual environment rather than using the native Windows operating system.
```
Comment (rsoeldner) [inline docs/smart-contracts/install/local-dev-node.md:158]: Is this the case for running Chainweb? I thought the WSL instructions are only for running a pact binary and chainweb always relies on docker (I might be wrong). @chessai

File: docs/smart-contracts/install/on-macos.md [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+import CodeBlock from '@theme/CodeBlock';
+
+# Install on macOS
+
+You can install Pact on macOS by using the Homebrew package manager or by downloading and installing the appropriate binary for the local system architecture. 
+
+## Prerequisites
+
+If you want to install Pact on macoS using the Homebrew package manager (recommended), you should verify that you have Homebrew installed by running the following command:
```
Comment (rsoeldner) [inline docs/smart-contracts/install/on-macos.md:16]:
```suggestion
If you want to install Pact on macOS using the Homebrew package manager (recommended), you should verify that you have Homebrew installed by running the following command:
```

File: docs/smart-contracts/install.md (hunk 1 above)
Comment (lsgunnlsgunn) [reply docs/smart-contracts/install.md:46]: Is there no plan to add the pact-cli to the development network image?
http://localhost:8080/ttyd/pact-cli/

File: docs/smart-contracts/install.md (hunk 2 above)
Comment (lsgunnlsgunn) [reply docs/smart-contracts/install.md:50]: To build Pact binaries directly from source, see the instructions in the appropriate repository:

- [Pact Core](https://github.com/kadena-io/pact-5/releases/tag/development-latest)
- [Pact 4.x](https://github.com/kadena-io/pact?tab=readme-ov-file#building-from-source)

File: docs/smart-contracts/install.md (hunk 3 above)
Comment (lsgunnlsgunn) [reply docs/smart-contracts/install.md:51]: In the immediate term, I guess we need to mention both (at least until Pact 5 goes to mainnet and replaces the current Pact.

**Pull request merged**